third_party/xla_client/pjrt_computation_client.cc

```diff
@@ -68,17 +78,17 @@ std::vector<ComputationClient::DataPtr> PjRtComputationClient::TransferToServer(
   std::vector<ComputationClient::DataPtr> datas;
   datas.reserve(tensors.size());
   for (auto& tensor : tensors) {
-    xla::Shape shape = tensor.shape;
-    xla::Literal literal(shape);
+    xla::Literal literal(tensor.shape);
     tensor.populate_fn(tensor, literal.untyped_data(), literal.size_bytes());
+    xla::Shape shape = LayoutUtil::GetWithDefaultLayout(tensor.shape);
```

Comment:
regarding layout I think we specified the layout if it is not optimal for TPU in 
https://github.com/pytorch/xla/blob/3c83269f8b163243263078c2f1e93d7617321712/torch_xla/csrc/layout_manager.cpp#L137-L152.

Always set the layout to default before sending to server seems incorrect to me.

Reply:
I agree. I looked at the `XrtComputationClient` to understand how the shape/layout was handled there, but it uses [`ShapeUtil::MakeShapeWithDescendingLayoutAndSamePhysicalLayout`](https://github.com/tensorflow/tensorflow/blob/cacc32bdef7f7f3d80e0780b63c76a9b75a3f3d4/tensorflow/compiler/xla/shape_util.h#L377), which actually changes the shape. I assume the shape is converted back somewhere else, but I couldn't find it.

https://github.com/pytorch/xla/blob/3c83269f8b163243263078c2f1e93d7617321712/third_party/xla_client/xrt_computation_client.cc#L1941

I assigned the default layout here because otherwise I would need to know the argument shapes at compile time (see [`CompileOptions.argument_layout`](https://github.com/tensorflow/tensorflow/blob/cacc32bdef7f7f3d80e0780b63c76a9b75a3f3d4/tensorflow/compiler/xla/pjrt/pjrt_client.h#L218)), and I don't think that information is passed to `ComputationClient::Compile` right now.

Reply:
I avoided the relayout here by disabling [`strict_shape_checking`](https://github.com/tensorflow/tensorflow/blob/b6725018e7ddc0258922998e9496d170b6485b68/tensorflow/compiler/xla/pjrt/pjrt_client.h#L855) in `Execute()` and added a TODO to set the argument layout explicitly.

Reply:
sounds good, if the real layout is required during compilation we should pass it down to the compile..